### ISSUE TITLE
group_concat section in select remains unformatted

### ISSUE THREAD

**Issue by csonuryilmaz** [manticore-projects/jsqlformatter#4]
Hi, below is a working unformatted MySQL query written in two lines:
```sql
SELECT GROUP_CONCAT(DISTINCT TRIM(Concat(m.title, ' ', m.firstname, ' ', m.lastname)) ORDER BY p2m.manufacturer_type_id ASC SEPARATOR ' ') as manufacturer_name
FROM product_to_manufacturer p2m LEFT JOIN manufacturer m ON m.manufacturer_id = p2m.manufacturer_id WHERE p2m.product_id = 574768;
```
When I format the query, `group_concat` section in select remains unformatted with all its keywords:
```sql
select GROUP_CONCAT(DISTINCT TRIM(Concat(m.title, ' ', m.firstname, ' ', m.lastname)) ORDER BY p2m.manufacturer_type_id ASC SEPARATOR ' ') as manufacturer_name
from product_to_manufacturer p2m
  left join manufacturer m
    on m.manufacturer_id = p2m.manufacturer_id
where p2m.product_id = 574768
;
```
Here is my formatter settings:
![2021-05-15_21-55](https://user-images.githubusercontent.com/7355362/118375190-bfbf3d00-b5c8-11eb-8a1c-5b164136a3a9.png)

---

Here is an example formatted query from an [online formatter service](https://www.freeformatter.com/sql-formatter.html): :wink: 
![2021-05-15_22-01](https://user-images.githubusercontent.com/7355362/118375241-38be9480-b5c9-11eb-952f-29494a40c76d.png)

**Comment by manticore-projects** [comment]
Thank you for reporting, the AGGREGATE functions are not fully supported yet because there is also an issue pending in JSQLParser which I wont' close before then other pending PRs are accepted.

That said: Also your Online example fails on it, the  GROUP_CONCAT is formatted somehow, but certainly not correctly.

But I will see what I can do today.

**Comment by manticore-projects** [comment]
You also will see some Warning like:
```
WARNING: Unhandled expression: net.sf.jsqlparser.expression.MySQLGroupConcat = GROUP_CONCAT(DISTINCT TRIM(Concat(m.title, ' ', m.firstname, ' ', m.lastname)) ORDER BY p2m.manufacturer_type_id ASC SEPARATOR ' ')
```

It indicates that this particular expression is not supported yet in JSQLFormatter.
Just send me a reminder whenever you any warning like that. I do not use MySQL and so far have focused on Oracle/MSSQL Server and H2. But of course we should support MySQL also as much as possible.

**Comment by manticore-projects** [comment]
How would you like to get it actually formatted? This MySQL specific expression looks a bit alien to me, so right now I would format it like that:
```sql
-- GROUP_CONCAT
SELECT Group_Concat(    DISTINCT Trim( Concat(  m.title, ' ', m.firstname
                                                , ' ', m.lastname ) )
                        ORDER BY p2m.manufacturer_type_id ASC
                        SEPARATOR' ' ) AS manufacturer_name
FROM product_to_manufacturer p2m
    LEFT JOIN manufacturer m
        ON m.manufacturer_id = p2m.manufacturer_id
WHERE p2m.product_id = 574768
;
```
Any objections or suggestions?

**Comment by manticore-projects** [comment]
Implemented, you can [test it online](http://217.160.215.75:8080/jsqlformatter/demo.html?args=-c%20LTAEHECUHkFUAUD6BhaA5ZBBAKgKAMoCiAMocthAE4D2ArgA4rUB2AxgIYAuAFKH6ABEAkvmxCMFbJQCWAW17IWHHn1kA6TtM4AbAKYAaUAHJjh9QDNplAM6dm7Wbtz8Xrt+4+e3hk0bNrtdlt7R1AASnDnL2i+aEgBQkhQACEATVB6ACZ1WXZmWnN2Vk5aSl1KRE4AT3pdRGkAE1BMfGQomK8ieExIHDjjAYiW0Fz8wuLS8sQQpwAxGABZDJoG2mLK6kRRgqKSssoM7PbQUlmKAClocRG8nYn9keOXdBG1bfG9qcbQAF5DnNuH0mFUauAA6gAJRKEf5qegrNacepNP4AVgA7AAWdEANgAHLgANxAA+) and also download the [Updated Netbeans Plugin](https://github.com/manticore-projects/jsqlformatter/releases/download/0.1.7-PRE/externalcodeformatter-1.14.1-SNAPSHOT.zip)

**Comment by csonuryilmaz** [comment]
> .. How would you like to get it actually formatted? This MySQL specific expression looks a bit alien to me, so right now I would format it like that: ..

Hi, your proposed formatting seems to be fine. :+1:

Now, I'm using latest `.nbm` and query is much better and readable compared to previous one-line style.

![2021-05-16_14-11](https://user-images.githubusercontent.com/7355362/118395067-1bcda400-b651-11eb-8ae0-9b9d4fe3751f.png)

 Thanks a lot. :hugs: :pray: 

> .. You also will see some Warning like:
> 
> WARNING: Unhandled expression: net.sf.jsqlparser.expression.MySQLGroupConcat 

I don't have any reported warning or exception on Netbeans "Notifications" window by now. 

> .. I do not use MySQL and so far have focused on Oracle/MSSQL Server and H2. But of course we should support MySQL also as much as possible. ..

:+1: In the past I've also worked on MSSQL and Oracle databases. But now, in my current job, I'm only working on MySQL.
